File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) 
    {
      int resultLocation = INT_MIN;
      int resultLocationMinDist = INT_MIN;
      vector<int> occupiedSeatLocations;
      for(int location=0; location<seats.size(); location++)
      {
        if(seats[location] == 1)
        {
          occupiedSeatLocations.push_back(location);
        }
      }
      
      
      int firstOccupiedSeatLocation = occupiedSeatLocations[0];
      int lastOccupiedSeatLocation = occupiedSeatLocations[occupiedSeatLocations.size() - 1];
      
      if(firstOccupiedSeatLocation != 0)
      {
        occupiedSeatLocations.insert(occupiedSeatLocations.begin(), -firstOccupiedSeatLocation);
      }
      
      if(lastOccupiedSeatLocation != seats.size() - 1)
      {
        int diff = seats.size() - 1 - lastOccupiedSeatLocation;
        occupiedSeatLocations.push_back(seats.size() - 1 + diff);
      }
      
      for(int index = 0; index < occupiedSeatLocations.size()-1; index++)
      {
        int currSeatLoc = occupiedSeatLocations[index];
        int nextSeatLoc = occupiedSeatLocations[index+1];
        int numberOfSeatsInBetween = nextSeatLoc - currSeatLoc;
        if(numberOfSeatsInBetween > 1)
        {
          int minDist = numberOfSeatsInBetween/2;
          int midSeatLocation = currSeatLoc + minDist;
          if(minDist > resultLocationMinDist)
          {
            resultLocationMinDist = minDist;
            resultLocation = midSeatLocation;
          }
        }
      }
      return resultLocationMinDist;
    }
};

File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.java
class Solution 
{
  public int maxDistToClosest(int[] seats) 
  {
    int count = 0, maxCount = Integer.MIN_VALUE;
    boolean isFirstOccupiedSeatPassed = true;
    for(int i=0;i<seats.length ;i++)
    {
      if(seats[i]==1)
      {
        if(isFirstOccupiedSeatPassed && i!=0)
          count *= 2;
        maxCount = Math.max(maxCount, count);
        count = 0;
        isFirstOccupiedSeatPassed = false;
      }
      else
        count++;
    }
    if(count>0)
      maxCount = Math.max(maxCount, 2*count);
    return (maxCount+1)/2;
  }
}